Kernel/Bus/VirtIO/Initializer.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/RefPtr.h>
+#include <AK/Result.h>
+#include <Kernel/Bus/VirtIO/Result.h>
+
+namespace Kernel::VirtIO {
+
+class Initializer {
+public:
+    static void detect();
+
+    template<typename DeviceType, typename... Args>
+    static inline Result<NonnullRefPtr<DeviceType>, VirtIO::InitializationResult> try_create_virtio_device(Args&&... args)
+    {
+        auto device = adopt_ref_if_nonnull(new DeviceType(forward<Args>(args)...));
```

Comment:
Not 100% sure about this one, since it's templated, but I think it is still missing a `(nothrow)`
```suggestion
        auto device = adopt_ref_if_nonnull(new (nothrow) DeviceType(forward<Args>(args)...));
```